### PR TITLE
Created OS class for OS detection and OS-baded commands

### PR DESCRIPTION
**Pull request by begedin**
A (hopefully) quick refactor of the OS detection code. It was going to be part of #9, but due to some complicated conflicts wiht #10, that part was merged, so we merge this separately.

### CODE REVIEW

**Comment by begedin** [comment]
I'm working on adding specs for this one.

File: lib/code/os.rb [hunk 1]
```diff
@@ -0,0 +1,37 @@
+module Code
+  module System
+    class OS
+
+      def self.current
+        new(platform_string: RUBY_PLATFORM)
+      end
```
Comment (begedin) [inline lib/code/os.rb]: Similar approach as with pull requests, branches and (soon) repositories.

This way, we can also inject the platform string for easier testing.

File: lib/code/os.rb [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+      def initialize(platform_string:)
+        @platform_string = platform_string
+      end
+
+      def windows?
```
Comment (venkatd) [inline lib/code/os.rb]: Great thinking with the methods.

File: spec/code/os_spec.rb [hunk 3]
```diff
@@ -0,0 +1,49 @@
+require 'code/os'
+
+module Code
+  describe "OS" do
+    it "detects linux" do
+      platform_string ="x86_64-linux"
+      os_detector = Code::System::OS.new(platform_string: platform_string)
```
Comment (venkatd) [inline spec/code/os_spec.rb]: The directory and module structures should map to one another. So this you would have `code/system/os.rb` and `spec/code/system/os_spec.rb`. Then in your test, you would have it like

``` ruby
module Code
    module System
        describe OS
        end
    end
end
```

Also this means you describe the OS class directly without using a string placeholder.

**Comment by venkatd** [comment]
This looks awesome.

File: spec/code/os_spec.rb [hunk 4]
```diff
@@ -0,0 +1,51 @@
+require 'code/os'
+
+module Code
```
Comment (venkatd) [inline spec/code/os_spec.rb]: Last change would be to make the directory structure match. So `code/system/os_spec.rb`

File: lib/code/os.rb (hunk 2 above)
Comment (idelahoz) [inline lib/code/os.rb]: This method is made specially for Nikola, LOL.

Comment (venkatd) [inline lib/code/os.rb]: Yeah maybe we should rename to `nikola?`

Comment (idelahoz) [inline lib/code/os.rb]: `nikola_is_that_you?`

File: spec/code/system/os_spec.rb [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+      it "detects unix" do
+        platform_string ="x86_64-linux"
+        os_detector = OS.new(platform_string: platform_string)
+
+        expect(os_detector.unix?).to be true
+      end
+
+      it "detects osx" do
+        platform_string = "x86_64-darwin14"
+        os_detector = Code::System::OS.new(platform_string: platform_string)
```
Comment (venkatd) [inline spec/code/system/os_spec.rb]: You can remove the namespaces here.

**Pull request merged**